Add documents about how to use ray/kuberay on aliyun (#52633)
ray-project/ray: doc/source/cluster/kubernetes/user-guides/ack-gpu-cluster.md

# Start Aliyun ACK Cluster with GPUs for KubeRay

This guide walks you through the steps to create an ACK cluster with GPU nodes specifically for KubeRay.
The configuration outlined here can be applied to most KubeRay examples found in the documentation.

## Step 1: Create a Kubernetes cluster on Aliyun ACK

Follow the first two steps in [this Aliyun documentation](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/create-an-ack-managed-cluster-2) to:
(1) create your Aliyun ACK cluster and (2) configure your computer to communicate with your cluster.

## Step 2: Create node pools for the Aliyun ACK cluster

Follow "Create a node pool" in [this Aliyun documentation](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/create-a-node-pool) to create node pools.

### Upgrade drivers on the nodes

If you need to upgrade the drivers on the nodes, follow "Step 2: Create a node pool and specify an NVIDIA driver version" in [this Aliyun documentation](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/customize-the-gpu-driver-version-of-the-node-by-specifying-the-version-number) to upgrade drivers.

> ## Step 3: Install Kuberay addon in the cluster

Reviewer: ```suggestion
## Step 3: Install KubeRay addon in the cluster
```

(kuberay-ack-gpu-cluster-setup)=

# Start Aliyun ACK Cluster with GPUs for KubeRay

This guide walks you through the steps to create an ACK cluster with GPU nodes specifically for KubeRay.
The configuration outlined here can be applied to most KubeRay examples found in the documentation.

## Step 1: Create a Kubernetes cluster on Aliyun ACK

> Follow the "Create a cluster" in [this Aliyun documentation](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/create-an-ack-managed-cluster-2) create your Aliyun ACK cluster and follow [this Aliyun documentation](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/access-clusters) to configure your computer to communicate with your cluster.

Reviewer 2: ```suggestion
See [Create a cluster](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/create-an-ack-managed-cluster-2) to create a Aliyun ACK cluster and see [Connect to clusters](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/access-clusters) to configure your computer to communicate with the cluster.
```

## Step 2: Create node pools for the Aliyun ACK cluster

> Follow "Create a node pool" in [this Aliyun documentation](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/create-a-node-pool) to create node pools.

Reviewer 2: ```suggestion
See [Create a node pool](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/create-a-node-pool) to create node pools.
```

### Manage node labels and taints

> If you need to set taints for nodes, follow "Create and manage node labels" and "Create and manage node taints" in [this Aliyun documentation](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/manage-taints-and-tolerations). For example, you can add a taint to gpu node pools so that ray head pods won't be scheduled to these nodes.

Reviewer 2: ```suggestion
If you need to set taints for nodes, see [Create and manage node labels](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/manage-taints-and-tolerations) and [Create and manage node taints](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/manage-taints-and-tolerations). For example, you can add a taint to GPU node pools so that Ray won't schedule head pods on these nodes.
```

### Upgrade drivers on the nodes

> If you need to upgrade the drivers on the nodes, follow "Step 2: Create a node pool and specify an NVIDIA driver version" in [this Aliyun documentation](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/customize-the-gpu-driver-version-of-the-node-by-specifying-the-version-number) to upgrade drivers.

Reviewer 2: ```suggestion
If you need to upgrade the drivers on the nodes, see [Step 2: Create a node pool and specify an NVIDIA driver version](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/customize-the-gpu-driver-version-of-the-node-by-specifying-the-version-number) to upgrade drivers.
```

## Step 3: Install KubeRay addon in the cluster

> Follow "Step 2: Install KubeRay-Operator" in [this Aliyun documentation](https://www.alibabacloud.com/help/en/ack/cloud-native-ai-suite/use-cases/efficient-deployment-and-optimization-practice-of-ray-in-ack-cluster?) to in KubeRay in ACK.

Reviewer 2: ```suggestion
See [Step 2: Install KubeRay-Operator](https://www.alibabacloud.com/help/en/ack/cloud-native-ai-suite/use-cases/efficient-deployment-and-optimization-practice-of-ray-in-ack-cluster?) to deploy KubeRay in ACK.
```

> (kuberay-ack-gpu-cluster-setup)=

Reviewer: I don't have access to Aliyun, so I can't verify whether this doc is easy to follow or not. Would you mind asking one of your colleagues to review this document?

Author: sure, thanks for suggestion
